infrastructure/client/api_client.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"taskmanager/domain/model"
+)
+
+type apiClient struct {
+	client    *http.Client
+	Token     string
+	ProjectId string
+}
+
+type ApiClient interface {
+	RequestAllTasks(t []*model.Task) ([]*model.Task, error)
+}
+
+func NewApiClient(token string, projectId string) ApiClient {
+	httpClient := &http.Client{}
+	return &apiClient{
+		client:    httpClient,
+		Token:     token,
+		ProjectId: projectId,
+	}
+}
+
+func (api *apiClient) RequestAllTasks(t []*model.Task) ([]*model.Task, error) {
```

Comment:
Just a question here: why aren't you handling different statuses returned by the API? i.e. 200, 400, etc.

Reply:
The main reason is because the API does not describe  the other possible status. I will add a condition to verify the Status Code is 200 which is the expected, if not I will log the Status Code and Response